src/mame/drivers/itech32.cpp

```diff
@@ -4920,6 +5009,18 @@ Label1  bne.s       Label1          ; Infinite loop if result isn't 0x80
 }
 
 
+void itech32_state::init_pubball()
+{
+	init_program_rom();
+	m_vram_height = 1024;
+	m_planes = 2;
+
+	m_maincpu->space(AS_PROGRAM).install_write_handler(0x300000, 0x300003, write8smo_delegate(*this, FUNC(itech32_state::color_w<0>)), 0x000000ff);
+	m_maincpu->space(AS_PROGRAM).install_write_handler(0x380000, 0x380003, write8smo_delegate(*this, FUNC(itech32_state::color_w<1>)), 0x000000ff);
+	m_maincpu->space(AS_PROGRAM).install_read_handler(0x180800, 0x180803, read32smo_delegate(*this, FUNC(itech32_state::trackball32_4bit_p1_r)));
```

Comment:
The color_w installations here seem redundant with the pubball memory map above? And if it has its own memory map, might as well chuck the input customizations there as well.

Reply:
They are, I wasn't aware that these changes were being made. I will rectify the redundancy sometime this weekend when I have time. For now, it's more important that we have support for the game in mainline for both visibility and consistency with the media push that is about to begin. Work with me, here.